### PR TITLE
Replace www-path with www-proxy option for development

### PR DESCRIPTION
**Pull request by anbraten**
By adding a new ENV variable called `WOODPECKER_WWW_PROXY` it is possible to serve a webinterface via a proxy configured by the `WOODPECKER_WWW_PROXY` value for development instead of serving the interface from the bundled code or from some folder location as the old `WOODPECKER_WWW` option allowed. Using a proxy allows developing the UI with hot-reloading.

image

### CODE REVIEW

File: cmd/drone-server/server.go [hunk 1]
```diff
@@ -83,13 +84,31 @@ func server(c *cli.Context) error {
...
-	// we are switching from gin to httpservermux|treemux,
-	// so if this code looks strange, that is why.
-	tree := setupTree(c)
+	proxyWebUI := c.String("www-proxy")
+
+	var webUIServe func(w http.ResponseWriter, r *http.Request)
+
+	if proxyWebUI == "" {
+		// we are switching from gin to httpservermux|treemux,
+		// so if this code looks strange, that is why.
```
Comment (6543) [inline cmd/drone-server/server.go:93]:
```suggestion
```

**Comment by 6543** [comment]
I'll have a closer look & test it once v0.14.0 got released - witch is only hold back by one issue

**Comment by 6543** [comment]
@anbraten if you have free cycles It would be nice if you could have a look at it - I wont have time until next week :/

**Comment by anbraten** [comment]
> @anbraten if you have free cycles It would be nice if you could have a look at it - I wont have time until next week :/

@6543 Do you mean this PR or #239 for 0.14.0. I can have a look at #239 anyways.
I already tested this PR and it should work without any problems.

File: router/router.go [hunk 2]
```diff
@@ -49,7 +48,8 @@ func Load(mux *httptreemux.ContextMux, middleware ...gin.HandlerFunc) http.Handl
 				session.User(c),
 			),
 		)
-		mux.ServeHTTP(c.Writer, req)
+
```
Comment (6543) [inline router/router.go:51]:
```suggestion
```

File: cmd/drone-server/server.go [hunk 3]
```diff
@@ -89,13 +90,30 @@ func server(c *cli.Context) error {
 	store_ := setupStore(c)
 	setupEvilGlobals(c, store_, remote_)
 
-	// we are switching from gin to httpservermux|treemux,
```
Comment (6543) [inline cmd/drone-server/server.go:92]: ... // we are switching from gin to httpservermux|treemux, ...

please do not remove - I think we should convert it into an TODO ...

Comment (anbraten) [reply cmd/drone-server/server.go:92]: I don't think it is a TODO, it sound more like a we are switching context from gin to httpservermux|treemux to me.

Comment (6543) [reply cmd/drone-server/server.go:92]: to be hosntest - it's guly and we only should use one router - so use gin for all and if it does not support all there are other drop in replacements

lets leave it out I'll create an issue!

**Review by 6543 — APPROVED** [review]

**Pull request merged**